nixos/tests/kernel-rust.nix

```diff
@@ -0,0 +1,30 @@
+{ pkgs, ... }: {
+  name = "kernel-rust";
+  meta = with pkgs.lib.maintainers; {
+    maintainers = [ blitz ];
+  };
+
+  nodes.machine = { config, pkgs, ... }:
+    {
+      boot.kernelPackages = pkgs.linuxPackages_6_6;
```

Comment:
nit: we should maybe use `latest` here because 6_6 might disappear at some point, or I can count on you updating it when it starts breaking.

Reply:
I used 6.6 specifically, because newer versions require finding the right patches and that would block updating `latest`. Since 6.6 is a LTS version, we should be good here for a long time and I expect Rust support for `latest` to be in a much better shape as 2024 progresses.

And yes, you can count on me fixing this if it breaks. :+1: